### PR TITLE
Update Fp3 to include fixes previously made to Fp and Fp2.

### LINKED ISSUE #1544: Update libsnark Fp3 to include sqrt fixes made for Fp and Fp2, or delete it
https://github.com/zcash/libsnark doesn't have its own issue tracker, so putting this here so I don't forget, even though we don't use Fp3::sqrt in Zcash.

### PR DESCRIPTION
**Pull request by syd0**
Also got rid of the #if DEBUG for consistency as mentioned
by @daira in a comment.

This closes #1544

### CODE REVIEW

File: src/snark/src/algebra/fields/fp3.tcc [hunk 1]
```diff
@@ -155,19 +159,6 @@ Fp3_model<n,modulus> Fp3_model<n,modulus>::sqrt() const
...
-    // check if square with euler's criterion
-    Fp3_model<n,modulus> check = b;
-    for (size_t i = 0; i < v-1; ++i)
-    {
-        check = check.squared();
-    }
-    if (check != one)
-    {
-        assert(0);
-    }
```
Comment (daira) [inline src/snark/src/algebra/fields/fp3.tcc]: This should be unconditional, not deleted: https://github.com/syd0/zcash/blob/master/src/snark/src/algebra/fields/fp2.tcc#L161-L170 . Also change ``assert`` to ``assert_except``.

**Comment by ebfull** [comment]
We should totally remove Fp3 if we aren't using it.

**Comment by daira** [comment]
I thought the class was used, just not the ``sqrt`` method? I could be wrong.

**Comment by ebfull** [comment]
I don't think we use it. We use Fp, Fp2, Fp6_3over2 and Fp12_2over3over2.

**Comment by daira** [comment]
Oh, then we should definitely delete it. @syd0, can you close this and open a PR to delete Fp3 instead?

**Comment by syd0** [comment]
Even better! Closing.

**Pull request closed without merging**